ros/api/v0/hosts.py

```diff
@@ -56,24 +56,20 @@ def get(self):
         limit = int(request.args.get('limit') or DEFAULT_HOSTS_PER_REP)
         offset = int(request.args.get('offset') or DEFAULT_OFFSET)
 
-        auth_key = request.headers.get('X-RH-IDENTITY')
-        if not auth_key:
-            response = make_response(
-                jsonify({"Error": "Authentication token not provided"}), 401)
-            abort(response)
-
-        inv_hosts = fetch_all_hosts_from_inventory(auth_key)
-        inv_host_ids = [host['id'] for host in inv_hosts['results']]
-
+        ident = identity(request)['identity']
         # Note that When using LIMIT, it is important to use an ORDER BY clause
         # that constrains the result rows into a unique order.
         # Otherwise you will get an unpredictable subset of the query's rows.
         # Refer - https://www.postgresql.org/docs/13/queries-limit.html
 
+        account_query = db.session.query(RhAccount.id).filter(RhAccount.account == ident['account_number']).subquery()
+        system_query = db.session.query(System.id)\
+            .filter(System.account_id.in_(account_query)).subquery()
+
         query = PerformanceProfile.query.filter(
```

Comment:
@semtexzv,
Instead of adding changes later to PR-21,
could you also address change from https://github.com/RedHatInsights/ros-backend/pull/21/files#diff-e303aa4e3e61a69895b0d295ce9166a08965a63a0c63569a4e1c8c60235f3b50R71 here?

I mean another subquery which will return only last reported record for particular system.

 ```
            last_reported = (
            db.session.query(PerformanceProfile.inventory_id, func.max(PerformanceProfile.report_date).label('max_date')
                             )
            .filter(PerformanceProfile.system_id.in_(system_query))
            .group_by(PerformanceProfile.system_id)
            .subquery()
        )
            
            query = (
            db.session.query(PerformanceProfile)
            .join(last_reported, (last_reported.c.max_date == PerformanceProfile.report_date) &
                  (PerformanceProfile.system_id == last_reported.c.system_id))
            .order_by(PerformanceProfile.id.asc())
        )
            
```